cpu6.c

```diff
@@ -1058,12 +1058,9 @@ static int dec(unsigned reg, unsigned val)
 static int clr(unsigned reg, unsigned v)
 {
 	reg_write(reg, v);
-	alu_out &= ~(ALU_F | ALU_L | ALU_M);
-	if (v == 0)
-		alu_out |= ALU_V;
-	else
-		/* Gets us past the tests but is probably wrong */
-		alu_out ^= ALU_V;
+	// Explicitly clears L and F flags
+	alu_out &= ~(ALU_L | ALU_F);
+	logic_flags(v);
```

Comment:
Would be neat if you describe CPU detection logic somewhere.

Reply:
As a comment?

Code just executes a clr to a non-zero value and checks the v flag. As CPU5 doesn't implement "small immediates", it clears to zero and v flag is zero.

Reply:
Aha, got it. CPU5 ignored the remaining nibble of the operand, where the constant was pushed in in cpu6.
Still perhaps worth leaving somewhere. In code comments or maybe on wiki.

Reply:
Ah so we finally know wtf was going on. Cool